OpenImis.ePayment/Data/ImisBasePayment.cs

```diff
@@ -803,11 +803,12 @@ public List<ReconciliationItem> ProvideReconciliationData(ReconciliationRequest
         public void CancelPayment(int payment_id)
         {
             var sSQL = @"UPDATE tblPayment
-                         SET PaymentStatus = 0
+                         SET PaymentStatus = @PaymentStatus
                          WHERE PaymentID = @PaymentID;";
 
             SqlParameter[] parameters = {
-                new SqlParameter("@PaymentID", payment_id)
+                new SqlParameter("@PaymentID", payment_id),
+                new SqlParameter("@PaymentStatus", PaymentStatus.Cancelled)
```

Comment:
@dragos-dobre btw I changed here in the cancellation to take from the static class values according to your tip from review.